src/main/java/com/microfocus/application/automation/tools/run/SseBuilder.java

```diff
@@ -180,13 +184,85 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher,
                 cdaDetails);
     	
         _sseModel.setAlmServerUrl(getServerUrl(_sseModel.getAlmServerName()));
-        
-        VariableResolver<String> varResolver = new VariableResolver.ByMap<String>(build.getEnvironment(listener));
-        Testsuites testsuites = execute(build, logger, varResolver);
-        
-        FilePath resultsFilePath = workspace.child(getFileName());
-        Result resultStatus = createRunResults(resultsFilePath, testsuites, logger);
-        provideStepResultStatus(resultStatus, build, logger);
+
+        // now merge them into one list
+        Properties mergedProperties = new Properties();
+
+        mergedProperties.putAll(_sseModel.getProperties());
+
+        mergedProperties.put("runType", AlmRunTypes.RunType.AlmLabManagement.toString());
+
+        Date now = new Date();
+        Format formatter = new SimpleDateFormat("ddMMyyyyHHmmssSSS");
+        String time = formatter.format(now);
+
+        // get a unique filename for the params file
+        ParamFileName = "props" + time + ".txt";
+        ResultFilename = "Results" + time + ".xml";
+        mergedProperties.put("resultsFilename", ResultFilename);
+
+        // get properties serialized into a stream
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        try {
+            mergedProperties.store(stream, "");
+        } catch (IOException e) {
+            listener.error("Storing run variable failed: " + e);
+            build.setResult(Result.FAILURE);
+        }
+        String propsSerialization = stream.toString();
+        InputStream propsStream = IOUtils.toInputStream(propsSerialization);
+
+        // get the remote workspace filesys
+        FilePath projectWS = workspace;
+
+        // Get the URL to the Script used to run the test, which is bundled
+        // in the plugin
+        URL cmdExeUrl =
+                Hudson.getInstance().pluginManager.uberClassLoader.getResource(HP_TOOLS_LAUNCHER_EXE);
```

Comment:
Hudson is a legacy library, why not use Jenkins?

Reply:
I guess that was the initial implementation when the construction of the plugin started.

Reply:
Changed to Jenkins